Java/libraries/recognizers-text-datatypes-timex-expression/src/main/java/com/microsoft/recognizers/datatypes/timex/expression/TimexHelpers.java

```diff
@@ -178,6 +179,31 @@ public static TimexProperty timexDateAdd(TimexProperty start, TimexProperty dura
         return start;
     }
 
+    public static String generateDateTimex(Integer year, Integer monthOrWeekOfYear, Integer day, Integer weekOfMonth, boolean byWeek)
+    {
+        String yearString = year == Constants.InvalidValue ? Constants.TimexFuzzyYear : TimexDateHelpers.fixedFormatNumber(year, 4);
+        String monthWeekString = monthOrWeekOfYear == Constants.InvalidValue ? Constants.TimexFuzzyMonth : TimexDateHelpers.fixedFormatNumber(monthOrWeekOfYear, 2);
+        String dayString;
+        if (byWeek)
+        {
+            dayString = day.toString(Integer.parseInt(Locale.ROOT.getLanguage()));
```

Comment:
Not quite sure about this: `toString(Integer.parseInt(Locale.ROOT.getLanguage()))`